Sources/_StringProcessing/Engine/Processor.swift

```diff
@@ -32,7 +32,7 @@ struct Processor<
   typealias Element = Input.Element
 
   let input: Input
-  let bounds: Range<Position>
+  var bounds: Range<Position>
```

Comment:
We have a bug or series of bugs that @natecook1000 is working on, but we probably won't want to be changing the bounds here.

Reply:
So we're fine with leaving the lower bound at the beginning of the string? I guess it still makes sense as the bounds of the string and we just need to set `currentPosition` to be where we want to start from when we reset?

Reply:
Right, though it might temporarily break some `firstMatch` tests until some anchors get fixed. @natecook1000 can you work with Lily here?